crates/runtime/src/http/routes.rs

```diff
@@ -74,13 +72,26 @@ pub(crate) fn routes(
             .layer(Extension(Arc::clone(&rt.embeds)));
     }
 
-    router = router
+    authenticated_router = authenticated_router
         .layer(Extension(Arc::clone(&rt.app)))
         .layer(Extension(Arc::clone(&rt.df)))
         .layer(Extension(Arc::clone(rt)))
         .layer(Extension(rt.metrics_endpoint))
         .layer(Extension(config));
-    router
+
+    // If we have an auth layer, add it to the authenticated router
+    if let Some(auth_layer) = auth_layer {
+        authenticated_router = authenticated_router.route_layer(auth_layer);
+    }
+
+    let unauthenticated_router = Router::new()
```

Comment:
We separate the unauthenticated routes here and only pass in the runtime object that powers the ready endpoint.